core/shared/src/main/scala/zio/prelude/recursive/Algebra.scala

```diff
@@ -0,0 +1,27 @@
+package zio.prelude.recursive
+
+import zio.prelude._
+
+sealed trait Algebra[Case[+_], A] extends (Case[A] => A) { self =>
```

Comment:
I'd suggest we call this `Folder`, because while it is an algebra, it is only an algebra in the sense of "algebraic data type", e.g. sum and product. Whereas, if we call it `Folder`, we can describe it as something that folds all the cases to an `A`.